### PR TITLE
Rename identifier_style and pattern_syntax

### PR DESCRIPTION
**Pull request by ddfreyne**
Old:

* `identifier_style` &rarr; `"full"` / `"stripped"`
* `pattern_syntax` &rarr; `"glob"` / `nil`

New:

* `identifier_type` &rarr; `"full"` / `"legacy"`
* `string_pattern_type` &rarr; `"glob"` / `"legacy"`

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ Changes:
 
 * Removed `ItemCollectionView#at`
 * Removed support for calling `ItemCollectionView#[]` with an integer
+* Renamed `identifier_type` to `identifier_type`, and made its values be `"full"` or `"legacy"`.
```
Comment (bobthecow) [inline NEWS.md]: > Renamed `identifier_type` to `identifier_type`

:P

Comment (ddfreyne) [inline NEWS.md]: Yup. I’m on to that right now. :)

**Comment by ddfreyne** [comment]
Hm, two things:

* The constructor of Nanoc::Identifier has an optional `:style` argument. It currently defaults to `"legacy'`, but that’s not what we want. Maybe having it mandatory (as `:type`) makes sense, or we default to `"full"`.

* `pattern_type` might be better as `string_pattern_type`. Regexes are also patterns, but `pattern_type` doesn’t apply to that.

File: lib/nanoc/base/source_data/identifier.rb [hunk 2]
```diff
@@ -16,10 +16,10 @@ def self.from(obj)
     end
 
     def initialize(string, params = {})
-      @style = params.fetch(:style, :stripped)
+      @style = params.fetch(:style, :legacy)
```
Comment (bobthecow) [inline lib/nanoc/base/source_data/identifier.rb]: should this variable be renamed as well?

Comment (ddfreyne) [inline lib/nanoc/base/source_data/identifier.rb]: Good catch.

**Comment by bobthecow** [comment]
I'd say name the argument (and instance variable) `type` and default it to `full`.

**Comment by ddfreyne** [comment]
Letting it default to `:full` seems like it belongs in a separate PR. I agree though.

Is this PR good to go?

**Comment by ddfreyne** [comment]
Oh yeah, hm, I would like to replace `pattern_type` with `string_pattern_type`.

**Comment by ddfreyne** [comment]
Renamed.

**Comment by ddfreyne** [comment]
Merged. Hope it’s good :)

**Pull request merged**